### PR TITLE
server/scheduler/evict_leader: fix a data race in the func `config.Clone`

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve?

Issue Number: Close [#4785](https://github.com/tikv/pd/issues/4785)

### What is changed and how does it work?
 Fix issue #4785

### Check List

Tests

- Unit test

### Release note

```release-note
 None.
```

### CODE REVIEW

File: server/schedulers/evict_leader_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 TiKV Project Authors.
```
Comment (disksing) [inline server/schedulers/evict_leader_test.go:1]: 2022

**Review by disksing — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**